src/libraries/System.IO.FileSystem/src/System/IO/DirectoryInfo.cs

```diff
@@ -74,20 +74,15 @@ public DirectoryInfo CreateSubdirectory(string path)
 
             string newPath = Path.GetFullPath(Path.Combine(FullPath, path));
 
-            ReadOnlySpan<char> trimmedNewPath = Path.TrimEndingDirectorySeparator(newPath.AsSpan());
-            ReadOnlySpan<char> trimmedCurrentPath = Path.TrimEndingDirectorySeparator(FullPath.AsSpan());
-
             // We want to make sure the requested directory is actually under the subdirectory.
-            if (trimmedNewPath.StartsWith(trimmedCurrentPath, PathInternal.StringComparison)
-                // Allow the exact same path, but prevent allowing "..\FooBar" through when the directory is "Foo"
-                && ((trimmedNewPath.Length == trimmedCurrentPath.Length) || PathInternal.IsDirectorySeparator(newPath[trimmedCurrentPath.Length])))
-            {
-                FileSystem.CreateDirectory(newPath);
-                return new DirectoryInfo(newPath);
-            }
-
-            // We weren't nested
-            throw new ArgumentException(SR.Format(SR.Argument_InvalidSubPath, path, FullPath), nameof(path));
+            // Allow the exact same path, but prevent allowing "..\FooBar" through when the directory is "Foo"
```

Comment:
Have you verified that this behavior is not being regressed?

Reply:
Best I can tell, that comment was added, and behavior fixed, here: 6c41cc9f889d3faa1ed03a8906632a0a5ad57b23
Which added or touched three tests:
`SubDirectoryIsParentDirectory_ThrowsArgumentException()`
`ValidPathWithTrailingSlash()`
`ValidPathWithoutTrailingSlash()`
All three of those do still pass. (At least on Linux. I've had only headaches trying to build this project on Windows so far, so haven't yet verified on that platform.)